packages/aardvark-react/src/default_landing_page.tsx

```diff
@@ -149,9 +206,28 @@ export class DefaultLanding extends React.Component<DefaultLandingProps, Default
 				onClick={ this.onStartAardvark }>Start Aardvark</div>;
 		}
 
+		if( this.state.hostSettings && this.state.manifest)
+		{
+			return <>
+				{ !this.state.hostSettings.favorited &&
+					<div className="LandingButton" onClick={ this.onAddFavorite }>Add to Favorites</div> }
+				{ this.state.hostSettings.favorited &&
+					<div className="LandingButton" onClick={ this.onRemoveFavorite }>Remove from Favorites</div> }
+				{ !this.state.hostSettings.markedForAutoLaunch && this.state.manifest.aardvark.startAutomatically &&
+					<div className="LandingButton" onClick={ this.onSetAutoLaunch}>Set to Auto Launch</div> }
+				{ this.state.hostSettings.markedForAutoLaunch && this.state.manifest.aardvark.startAutomatically &&
+					<div className="LandingButton" onClick={ this.onRemoveAutoLaunch}>Remove from Auto Launch Gadgets</div> }
+				<div className="LandingButton" onClick={ this.onStartGadget }>Start Gadget</div>
+				{ this.state.addResult &&
+					<div style={ { fontSize: "medium" }}>{ this.state.addResult }</div> }
+				</>;
```

Comment:
There's a lot of duplication with this and the block below. Maybe these should use nullish coalescing?

`this.state.hostSettings?.favorited && <blah/>`

Then you could just mention them once.